azure-application-insights-spring-boot-starter/src/main/java/com/microsoft/applicationinsights/boot/ApplicationInsightsWebMvcAutoConfiguration.java

```diff
@@ -0,0 +1,76 @@
+ * publish, distribute, sublicense, and/or sell copies of the Software, and to permit
+ * persons to whom the Software is furnished to do so, subject to the following conditions:
+ * The above copyright notice and this permission notice shall be included in all copies or
+ * substantial portions of the Software.
+ * THE SOFTWARE IS PROVIDED *AS IS*, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED,
+ * INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR
+ * PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE
+ * FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR
+ * OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+ * DEALINGS IN THE SOFTWARE.
+ */
+
+package com.microsoft.applicationinsights.boot;
+
+import com.microsoft.applicationinsights.TelemetryConfiguration;
+import com.microsoft.applicationinsights.web.internal.WebRequestTrackingFilter;
+import com.microsoft.applicationinsights.web.spring.internal.InterceptorRegistry;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.boot.autoconfigure.AutoConfigureAfter;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnBean;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnMissingBean;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnWebApplication;
+import org.springframework.boot.web.servlet.FilterRegistrationBean;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.context.annotation.DependsOn;
+import org.springframework.context.annotation.Import;
+import org.springframework.core.Ordered;
+
+/**
+ * <h1>Configuration for Auto-collection of HTTP requests.</h1>
+ *
+ * <p>
+ *   This class is responsible for configuring {@link WebRequestTrackingFilter} for auto collection
+ *   of incoming HTTP requests
+ * </p>
+ *
+ * @author Arthur Gavlyukovskiy
+ */
+
+@Configuration
+@Import(InterceptorRegistry.class)
+@ConditionalOnBean(TelemetryConfiguration.class)
+@ConditionalOnWebApplication
+@ConditionalOnProperty(value = "azure.application-insights.web.enabled", havingValue = "true", matchIfMissing = true)
+@AutoConfigureAfter(ApplicationInsightsTelemetryAutoConfiguration.class)
+public class ApplicationInsightsWebMvcAutoConfiguration {
+
+    @Bean
+    public FilterRegistrationBean webRequestTrackingFilterRegistrationBean(WebRequestTrackingFilter webRequestTrackingFilter) {
+        FilterRegistrationBean registration = new FilterRegistrationBean();
+        registration.setFilter(webRequestTrackingFilter);
+        registration.addUrlPatterns("/*");
+        registration.setName("webRequestTrackingFilter");
```

Comment:
This should not be hardcoded.

Reply:
@littleaj and @grlima I looked into this method. This is method is usually used to set the name of the RegistrationBean, however it is not necessary to set the name and the default behavior here is that if this is not set explicitly the name of the bean would be taken into the account. 

Further, a surprising thing this method no longer seems to be present in the latest SpringBoot framework however no mention anywhere in the change logs too. 

https://github.com/spring-projects/spring-boot/commit/9cb5f3da89b26cc60725a7e0a0c300a53d38f91f#diff-4c7df2b424db8ecb8e18ac9485ddcc8d

Therefore just to be safe i would remove it. Anyways, it doesn't affect the behavior either.

Reply:
makes sense

Reply:
Done